TU-Agent/tygron.mod2g

```diff
@@ -9,16 +9,23 @@ exit=nogoals.
 order=linear.
 
 module tygron {
-
-	% Call the module buildConstruction when there is a multipolygon available.
-	if bel(demolishedPolygon(MP)) 
-		then buildConstruction.
 	
-	% Call the module demolish to demolish EWI
-	if not(bel(demolishedPolygon(MP)))
+	%TODO budget?
+	
+	%TODO check for requestgoals and handle these first so other stakeholders can move on without waiting on our bot, prevents a stalemate%
+	
+	%if the buildConstruction module adopted the goal to create land we should enter the demolish module
+	%TODO: We could also buy land to create buildable land
+	if goal(createLandToBuild)
 		then demolish.
 	
-	% if we have no room to build we should demolish something or buy land
-    % then demolish.
-	% then buyLand.
+	%If we have the goal to build high buildings, we should enter the build module
+	if goal(indicatorgoal('Bouwen hoge gebouwen', _))
+		then buildConstruction.
+	
+	%If the goal to build high buildings is met but we still have the demolish goal, we should enter the demolish module
+	if goal(indicatorgoal('Slopen oud gebouwen', _))
```

Comment:
If we change them in the engine and in the code, sure. But since we're working with duplicates of the map we would have to change them in every map, we can, but i don't think it should have priority now.